src/components/TaskDetail.jsx

```diff
@@ -0,0 +1,312 @@
+                  <span>{label.text}</span>
+                  {user && (
+                    <button
+                      className="remove-label-button"
+                      onClick={() => onRemoveLabel(task.id, label.id)}
+                      title="Remove label"
+                    >
+                      ×
+                    </button>
+                  )}
+                </div>
+              ))}
+              {(!task.labels || task.labels.length === 0) && (
+                <div className="no-labels-message">
+                  No labels added yet
+                </div>
+              )}
+            </div>
+          </div>
+
+          <div className="task-description">
+            <div className="description-header">
+              <h3>Description</h3>
+              {user && !showDeleteConfirm && (
+                <button 
+                  className="edit-button"
+                  onClick={() => setIsEditing(!isEditing)}
+                >
+                  {isEditing ? 'Cancel' : 'Edit'}
+                </button>
+              )}
+            </div>
+
+            {user && isEditing && !showDeleteConfirm ? (
+              <div className="description-editor" data-color-mode={isDark ? "dark" : "light"}>
+                <MDEditor
+                  value={description}
+                  onChange={setDescription}
+                  preview="live"
+                  height={200}
+                  hideToolbar={false}
+                  enableScroll={true}
+                  style={{
+                    backgroundColor: isDark ? '#27272A' : '#ffffff',
+                  }}
+                  previewOptions={{
+                    rehypePlugins: [[rehypeSanitize]],
+                  }}
+                />
+                <div className="editor-actions">
+                  <button 
+                    className="save-button"
+                    onClick={handleSave}
+                  >
+                    Save
+                  </button>
+                </div>
+              </div>
+            ) : (
+              <div className="description-preview" data-color-mode={isDark ? "dark" : "light"}>
+                <MDEditor.Markdown 
+                  source={description || 'No description provided.'} 
+                  rehypePlugins={[[rehypeSanitize]]}
+                  style={{
+                    backgroundColor: 'transparent',
+                    color: 'inherit'
+                  }}
+                />
+              </div>
+            )}
+          </div>
+
+          <div className="task-metadata">
+            <div className="metadata-item">
+              <span className="metadata-label">Created:</span>
+const formatCreatedAt = (timestamp) => {
+  if (!timestamp) return 'Unknown';
+  try {
+    const date = timestamp.seconds
+      ? new Date(timestamp.seconds * 1000)
+      : new Date(timestamp);
+    return date.toLocaleString();
+  } catch (error) {
+    return 'Invalid date';
+  }
+};
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
// Em um arquivo utils/dateFormatters.js
export const formatCreatedAt = (timestamp) => {
```
A função formatCreatedAt pode ser movida para fora do componente para evitar recriação desnecessária a cada renderização. Considere movê-la para um arquivo de utilitários separado.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;